crates/evm/src/evm/executor.rs

```diff
@@ -101,29 +103,40 @@ pub(crate) fn execute_multiple_tx<C: sov_modules_api::Context, EXT: CitreaExtern
                 .entered();
 
         if tx.signer() == SYSTEM_SIGNER {
-            let shp_provider = SHORT_HEADER_PROOF_PROVIDER
-                .get()
-                .expect("Short header proof provider not set");
-            // TODO: Check if this is the set_block_info tx, if so:
-            // TODO: Get this and other params from input
-            // TODO: don't forget about height
-            // TODO: read height - 1's hash from the contract, and compare
-            // with the prevhash result from the short header proof
-            // reject L2 block if they don't match
-            let _input = tx.input();
-            let l1_hash = [0u8; 32];
-            match shp_provider.get_and_verify_short_header_proof_by_l1_hash(l1_hash) {
-                Ok(true) => {}
-                Ok(false) => {
-                    // Failed to verify shp
-                    return Err(SoftConfirmationModuleCallError::ShortHeaderProofVerificationError);
-                }
-                Err(ShortHeaderProofProviderError::ShortHeaderProofNotFound) => {
-                    return Err(SoftConfirmationModuleCallError::ShortHeaderProofNotFound);
+            if citrea_spec < CitreaSpecId::Fork2 {
+                native_error!("System transaction found in user txs");
+                return Err(SoftConfirmationModuleCallError::EvmMisplacedSystemTx);
+            }
+
+            let function_selector: [u8; 4] = tx.input()[0..4]
```

Comment:
There is a funky way to parse this data using BitcoinLightClientContract generated fns from abi